### ISSUE TITLE
Two Xbox Types files Conflict

### ISSUE THREAD

**Issue by RadWolfie** [Cxbx-Reloaded/Cxbx-Reloaded#1953]
## Quick summary
`src/common/xbox_types.h` vs `src/common/xbox/Types.hpp` are conflict for naming both file and path.

## Details
I recently discover there are two xbox types naming within our common's folder which are `src/common/xbox_types.h` and `src/common/xbox/Types.hpp`. `xbox_types.h` header file should reside inside xbox folder within common's folder.

In order to resolve this conflict, I suggest to rename Types._pp files to something else and move xbox_types.h header file into xbox folder as `types.h` file name. Currently, I'm not sure what name it should follow as since it contain both system and xbe type references.

### System Configuration:
N/A

### Additional Information (if any):
N/A

**Comment by ergo720** [comment]
`Types.hpp` deals with xbe types, and for that we already have the folder `src/common/xbe`, which is more appropriate in my opinion. So I instead suggest to move `Types.hpp` and `Types.cpp` in the xbe folder instead and delete the folder `src/common/xbox`. Some possible new names for the files would be `xbe_types.hpp` and `xbe_types.cpp`. The only issue left is then where to put the function `GetSystemTypeToStr`, since that's not strictly related to the xbe functionality.

**Comment by RadWolfie** [comment]
I disagree with removal of `src/common/xbox` since it is unique common folder designated for any work done relative to xbox's structures and coding that should not be in the core folder. If you have looked in my drafted pull request contain logging files in xbox folder since both LLE and HLE can have same structures to share together. There are chances over time we will have more xbox relative files in that folder to be honestly.

P.S. Over time we will be removing messy includes to enforce follow relative or absolute file path in the includes.

**Comment by RadWolfie** [comment]
`src/common/xbe` folder and **AddressRanges**, from common folder, files may will be move into `src/common/xbox` folder along with **XADPCM.h**, and **xbox_types.h**. Common folder, root directory in src folder, should be focus on interfaces without dependent on specific platform's header files.

Off topic, we should also move win32 and linux into platform folder as well. Since they are dependency on platform requirement in order to work correctly.
